homeassistant/components/thethingsnetwork/coordinator.py

```diff
@@ -0,0 +1,149 @@
+"""The Things Network's integration DataUpdateCoordinator."""
+
+from collections.abc import Callable
+from datetime import timedelta
+import logging
+import traceback
+from typing import TYPE_CHECKING
+
+from ttn_client import TTNAuthError, TTNClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    CONF_API_KEY,
+    CONF_APP_ID,
+    CONF_HOSTNAME,
+    DEFAULT_API_REFRESH_PERIOD_S,
+    DEFAULT_FIRST_FETCH_LAST_H,
+    OPTIONS_MENU_EDIT_INTEGRATION,
+    OPTIONS_MENU_INTEGRATION_FIRST_FETCH_TIME_H,
+    OPTIONS_MENU_INTEGRATION_REFRESH_TIME_S,
+)
+from .entry_settings import TTN_EntrySettings
+
+if TYPE_CHECKING:
+    from .entity import TTN_Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TTNCoordinator(DataUpdateCoordinator):
+    """TTN coordinator."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            # Name of the data. For logging purposes.
+            name=f"TheThingsNetwork_{entry.data[CONF_APP_ID]}",
+            # Polling interval. Will only be polled if there are subscribers.
+            update_interval=timedelta(
+                seconds=entry.options.get(OPTIONS_MENU_EDIT_INTEGRATION, {}).get(
+                    OPTIONS_MENU_INTEGRATION_REFRESH_TIME_S,
+                    DEFAULT_API_REFRESH_PERIOD_S,
+                )
+            ),
+        )
+        self.__entry: ConfigEntry = entry
+        self.__entity_class_register: list[TTNCoordinator.__RegisteredEntityClass] = []
+
+        self.__client = TTNClient(
+            entry.data[CONF_HOSTNAME],
+            entry.data[CONF_APP_ID],
+            entry.data[CONF_API_KEY],
+            self.__entry.options.get(OPTIONS_MENU_EDIT_INTEGRATION, {}).get(
+                OPTIONS_MENU_INTEGRATION_FIRST_FETCH_TIME_H, DEFAULT_FIRST_FETCH_LAST_H
+            ),
```

Comment:
What's this?

Reply:
before removing the options menu, this was used to allow configuring the refresh value. Since the feedback was to remove the support to configure it then I will remove this option.